api/proto/scheduler/scheduler.proto

```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+
+enum RunnerType {
+  DOCKER = 0;
+}
+
+message Action {
+  string action_id = 1;
+  RunnerType runner_type = 2;
+  optional string image = 3;
+  repeated string commands = 4;
+}
+
+message RegisteredAgent {
+  string hostname = 1;
+  int32 cpu = 2;
+  int32 memory = 3;
+}
+
+message AgentId {
+  string id = 1;
+}
+
+message ActionLog {
+  string action_id = 1;
+  string log = 2;
+  optional bool success = 3; // Only populated if action is complete
+  optional int32 error_code = 4; // Only populated if success is false
```

Comment:
Maybe this is redundant. I would have implement only an exit_code. I saw that if the code is different from 0 you can consider that you action failed (maybe this has to be confirmed)

Reply:
I would suggest for this message to be named `ActionLog` and for it to include a nested `ActionResult`message:

```proto
enum CompletionType  {
    SUCCESS = 0;
    FAILURE = 1;
    CANCELLED = 2;
    TIMEOUT = 3;
}

message ActionResult {
   CompletionType completion = 1;
   optional int32 exit_code = 2;
}


```

Reply:
changes done